### ISSUE TITLE
GenomicsDB should support multi-nucleotide representations

### ISSUE THREAD

**Issue by ldgauthier** [broadinstitute/gatk#6500]
We don't support converting adjacent SNPs into MNPs for GVCFs, but in some cases a haplotype will be described as adjacent insertions and deletions that lead to a multinucleotide reference allele with an alternate allele longer than one base as in #6473  Whereas a lot of my concerns about MNPs in GVCFs arise from scenarios where the variants are phased in trans, these alternate bases should always be in cis since they were discovered on the same haplotype.

For example, such a haplotype might be aligned to the reference as `102M4D4I300M` which get converted to an event like GGGCA -> GTTGT.  This is different from the trans phased scenario because there will only be one alt in the single-sample GVCF, whereas the trans example would be a 1/2 genotype.  I still want to reject the 1/2 trans phased MNPs because they will complicate joint calling, but for a single sample with a single multinucleotide alt allele, it should behave nearly the same way as a deletion.

See also https://gatk.broadinstitute.org/hc/en-us/community/posts/360058358051-HaplotypeCaller-Bad-input-GenomicsDBImport-does-not-support-GVCFs-with-MNPs

**Comment by kgururaj** [comment]
Hello Laura, one question. You stated that the above alt allele should be treated as a deletion. Just wanted to check with you a boolean condition to determine if an alt allele is a deletion (not including symbolic allele handling for simplicity):
```cpp
is_deletion = ( ref_allele.length() > alt_allele.length() )
    || ( ref_allele.length() == alt_allele.length() 
        && ref_allele[0] == alt_allele[0] )
```
The second predicate in the OR expression is for the case described above. Do you think this condition is correct?

**Comment by mlathara** [comment]
@ldgauthier we talked about this during the GATK office hours meeting today, and it was suggested that something like this might be an appropriate boolean condition for checking whether an alt allele is a deletion:

```
is_deletion = (ref_allele.length() > alt_allele.length())
    || (ref_allele.length() == alt_allele.length() && mismatch_base_count(ref_allele, alt_allele) > 1)
```

where `mismatch_base_count` is a function that counts the number of bases that don't match between the REF and ALT. Does that sound good to you?

**Comment by droazen** [comment]
@mlathara We were actually trying to craft a boolean expression to detect MNPs, not deletions. The one we came up with was: ref and alt are the same length, the length is > 1, and the number of mismatching bases is > 1. The last condition (mismatching bases > 1) is necessary to rule out padded SNPs.

**Comment by mlathara** [comment]
@droazen Right - so the reason the pseudo code is formulated as something to detect deletions is because Laura had mentioned in the original ticket (and in the meeting we had last week) that she thought we could handle this case like it was a deletion.  Quoting from her ticket:

> I still want to reject the 1/2 trans phased MNPs because they will complicate joint calling, but for a single sample with a single multinucleotide alt allele, it should behave nearly the same way as a deletion.

So, yes the second part of the OR expression (we dropped length > 1, since that will be the case if number of mismatching bases > 1) detects MNP. And if the MNPs Laura describes above should be treated as deletions this may be an easy way to do so.

**Comment by kgururaj** [comment]
One more question for @ldgauthier 

>  because they will complicate joint calling, but for a single sample with a single multinucleotide alt allele, it should behave nearly the same way as a deletion.

does this mean that these MNPs should produce spanning deletions at subsequent positions?

**Comment by ldgauthier** [comment]
Good point. Yes, a sample with a MNP should get a symbolic allele in it's genotype for any variants overlapping the MNP reference allele.  The * allele is definitely specifically in the VCF 4.2 spec as resulting from a deletion, so I think we'll have to do something else, which is a bummer.  I'm not feeling very creative at the moment.  @droazen @lbergelson any ideas?

**Comment by lbergelson** [comment]
We're going to redefine * as any spanning allele instead of just deletions.  I don't know if it will technically change in 4.2, but I think it's ok to use * for other overlapping upstream stuff.

**Comment by kgururaj** [comment]
One more clarification question: Assume there is a sample which has both a deletion and an MNP as ALT alleles at a given position. For deletions, we use the min PL value to determine which ALT allele represents the spanning deletion in the subsequent positions. Should we combine the PL values of MNPs and deletions?

**Comment by ldgauthier** [comment]
It's not quite as logical to combine MNPs and deletions as it is to combine two deletions, but given that we only have one "symbolic" allele to represent both, I don't think there's any other choice.

**Comment by droazen** [comment]
@nalinigans This one is done, isn't it? Can this ticket be closed?

**Comment by nalinigans** [comment]
Yes, this can be closed.

**Comment by droazen** [comment]
Closing -- implemented in GATK 4.1.8.0
